### PR TITLE
ci: update optimize snapshot and latest docker tags

### PR DESCRIPTION
**Pull request by OmranAbazid**
## Description

See https://github.com/camunda/camunda-optimize/issues/13498

### CODE REVIEW

**Comment by OmranAbazid** [comment]
We need to adjust this PR to respect the changes done in:
https://github.com/camunda/camunda/pull/20613

**Review by RomanJRW — CHANGES_REQUESTED** [review]
In the pom, there is a section for cleaning up the docs linked, marked with: `<!-- Remove -SNAPSHOT, -rcX, and -alphaX from docs version if it exists -->`. Does the logic here need adjusting too?

File: .github/optimize/scripts/build-docker-image.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ if [ "${PUSH_LATEST_TAG}" = "true" ]; then
 fi
 
 if [ "${IS_MAIN}" = "true" ]; then
-    tags+=("${DOCKER_IMAGE_DOCKER_HUB}:SNAPSHOT")
+    tags+=("${DOCKER_IMAGE_DOCKER_HUB}:8-SNAPSHOT")
```
[the remark this answers is not in the document]
Comment (OmranAbazid) [reply .github/optimize/scripts/build-docker-image.sh:12]: Not sure what you mean here.
I think the snapshot docker tag is not preceded with a version number.
When pulling the snapshot, we used to pull it like this: `docker image pull camunda/optimize:SNAPSHOT`

If we keep it as it is now, both optimize 7 and optimize 8 will keep overriding the image.

**Review by RomanJRW — APPROVED** [review]
Approve here, as long as the dependencies listed on the issue are also taken care of 👍

**Comment by OmranAbazid** [comment]
> In the pom, there is a section for cleaning up the docs linked, marked with: " Remove -SNAPSHOT, -rcX, and -alphaX from docs version if it exists ". Does the logic here need adjusting too?

I think there is a difference between the snapshot version of the docker image and the snapshot development version.
I only adjusted the snapshot docker image, but the snapshot development version is still the same.

Therefore, I think we do not need to update the regex logic.

**Pull request merged**